egs/lre07/v1/lid/train_ivector_extractor_dnn.sh

```diff
@@ -0,0 +1,180 @@
+# Apache 2.0.
+
+# This script trains the i-vector extractor using a DNN-based UBM. It also requires
+# an fGMM, created by the script lid/init_full_gmm_from_dnn.sh.
+# Note: there are 3 separate levels of parallelization: num_threads, num_processes, 
+# and num_jobs.  This may seem a bit excessive.  It has to do with minimizing 
+# memory usage and disk I/O, subject to various constraints.  The "num_threads" 
+# is how many threads a program uses; the "num_processes" is the number of separate
+# processes a single  job spawns, and then sums the accumulators in memory.
+# Our recommendation:
+#  - Set num_threads to the minimum of (4, or how many virtual cores your machine has).
+#    (because of needing to lock various global quantities, the program can't
+#    use many more than 4 threads with good CPU utilization).
+#  - Set num_processes to the number of virtual cores on each machine you have, divided by 
+#    num_threads.  E.g. 4, if you have 16 virtual cores.   If you're on a shared queue
+#    that's busy with other people's jobs, it may be wise to set it to rather less
+#    than this maximum though, or your jobs won't get scheduled.  And if memory is
+#    tight you need to be careful; in our normal setup, each process uses about 5G.
+#  - Set num_jobs to as many of the jobs (each using $num_threads * $num_processes CPUs)
+#    your queue will let you run at one time, but don't go much more than 10 or 20, or
+#    summing the accumulators will possibly get slow.  If you have a lot of data, you
+#    may want more jobs, though.
+
+# Begin configuration section.
+nj=10   # this is the number of separate queue jobs we run, but each one 
+        # contains num_processes sub-jobs.. the real number of threads we 
+        # run is nj * num_processes * num_threads, and the number of
+        # separate pieces of data is nj * num_processes.
+num_threads=4
+num_processes=4 # each job runs this many processes, each with --num-threads threads
+cmd="run.pl"
+stage=-4
+num_gselect=20 # Gaussian-selection using diagonal model: number of Gaussians to select
+ivector_dim=400 # dimension of the extracted i-vector
+use_weights=false # set to true to turn on the regression of log-weights on the ivector.
+num_iters=10
+min_post=0.025 # Minimum posterior to use (posteriors below this are pruned out)
+num_samples_for_weights=3 # smaller than the default for speed (relates to a sampling method)
+cleanup=true
+posterior_scale=1.0 # This scale helps to control for successve features being highly
+                    # correlated.  E.g. try 0.1 or 0.3
+sum_accs_opt=
+# End configuration section.
+
+echo "$0 $@"  # Print the command line for logging
+
+if [ -f path.sh ]; then . ./path.sh; fi
+. parse_options.sh || exit 1;
+
+
+if [ $# != 5 ]; then
+  echo "Usage: $0 <fgmm-model> <dnn-model> <data-speaker-id> <data-dnn> <extractor-dir>"
+  echo " e.g.: $0 exp/sup_ubm/final.ubm exp/dnn/final.mdl data/train data/train_dnn exp/extractor_male"
```

Comment:
I think this should be updated to remove the reference to extractor_male, since that is SID specific